api/v1/aerospikecluster_validating_webhook.go

```diff
@@ -1396,6 +1392,47 @@ func validateAerospikeConfigUpdate(
 	return validateNsConfUpdate(incomingSpec, outgoingSpec, currentStatus)
 }
 
+func validateTLSUpdate(oldConf, newConf map[string]interface{}) error {
+	oldTLS, ok11 := oldConf["network"].(map[string]interface{})["tls"]
+	newTLS, ok22 := newConf["network"].(map[string]interface{})["tls"]
+
+	if ok11 && ok22 && (!reflect.DeepEqual(oldTLS, newTLS)) {
+		oldTLSCAFileMap := make(map[string]string)
+		usedTLS := sets.NewString()
+
+		// fetching names of TLS configurations used in connections
+		for _, connectionType := range networkConnectionTypes {
+			if connectionConfig, exists := newConf["network"].(map[string]interface{})[connectionType]; exists {
+				connectionConfigMap := connectionConfig.(map[string]interface{})
+				if tlsName, ok := connectionConfigMap[confKeyTLSName]; ok {
+					usedTLS.Insert(tlsName.(string))
+				}
+			}
+		}
+
+		for _, tls := range oldTLS.([]interface{}) {
+			tlsMap := tls.(map[string]interface{})
+
+			oldCAFile, oldCAFileOK := tlsMap["ca-file"]
+			if oldCAFileOK {
+				oldTLSCAFileMap[tlsMap["name"].(string)] = oldCAFile.(string)
+			}
+		}
```

Comment:
I think this oldTLSCAFileMap should be populated for only those TLS which were used earlier in any sub-context. If TLS is just added and not used then we can allow any type of update on that